### PR TITLE
privilege: add DDL and DML privilege check for system tables

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve? 

Add DDL and DML privilege check for system tables in performance_schema and metrics_schema.

### What is changed and how it works?

Prevent users from executing `ALTER`, `DROP`, `INDEX`, `INSERT`, `UPDATE`, `DELETE` statements on predefined tables.
Privileges of those tables which are defined in these schema by users themselves are kept untouched.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
 ```
mysql> delete from events_statements_summary_by_digest;
ERROR 8121 (HY000): privilege check fail
```

Code changes

 - Has exported function/method change

Side effects

 - Breaking backward compatibility

Related changes

N/A

Release note

 - Forbid users to execute DDL and update/delete/insert predefined tables in performance_schema and metrics_schema.
- Compatibility declaration:  Executing DDL and update/delete/insert predefined tables in performance_schema and metrics_schema are not allowed any longer.

### CODE REVIEW

File: privilege/privileges/privileges.go [hunk 1]
```diff
@@ -51,15 +53,30 @@ func (p *UserPrivileges) RequestVerification(activeRoles []*auth.RoleIdentity, d
...
 		}
 		return true
+	// We should be very careful of limiting privileges, so ignore `mysql` for now.
+	case util.PerformanceSchemaName.L, util.MetricSchemaName.L:
+		// CREATE and DROP privileges are not limited in the older versions, so ignore them now.
+		// User may have created some tables in these schema, but predefined tables can't be altered or modified.
+		if (dbLowerName == util.PerformanceSchemaName.L && perfschema.IsPredefinedTable(table)) ||
+			(dbLowerName == util.MetricSchemaName.L && infoschema.IsMetricTable(table)) {
+			switch priv {
+			case mysql.AlterPriv, mysql.DropPriv, mysql.IndexPriv, mysql.InsertPriv, mysql.UpdatePriv, mysql.DeletePriv:
```
Comment (AilinKid) [inline privilege/privileges/privileges.go]: how about `CREATE PRIV`?

File: infoschema/perfschema/tables_test.go [hunk 2]
```diff
@@ -65,6 +66,11 @@ func (s *testTableSuite) TearDownSuite(c *C) {
 	s.store.Close()
 }
 
+func (s *testTableSuite) TestPredefinedTables(c *C) {
+	c.Assert(perfschema.IsPredefinedTable("EVENTS_statements_summary_by_digest"), IsTrue)
```
Comment (AilinKid) [inline infoschema/perfschema/tables_test.go]: Better put it into other suitable test case if there are only two lines for this function in the future.

File: privilege/privileges/privileges.go (hunk 1 above)
Comment (djshow832) [reply privilege/privileges/privileges.go]: I have comments in this function.
If running applications with an older version happen to create table in this schema, but creating table is forbidden after upgrading, it'll cause an incident.

File: infoschema/perfschema/tables_test.go (hunk 2 above)
Comment (djshow832) [reply infoschema/perfschema/tables_test.go]: No other suitable test case in this file now.

File: privilege/privileges/privileges.go (hunk 1 above)
Comment (AilinKid) [reply privilege/privileges/privileges.go]: for compatibility, ok

**Pull request merged**